internal/operands/vm-console-proxy/reconcile.go

```diff
@@ -271,10 +331,43 @@ func getVmConsoleProxyImage() string {
 	return common.EnvOrDefault("VM_CONSOLE_PROXY_IMAGE", "quay.io/kubevirt/vm-console-proxy:v0.1.0")
 }
 
+func findResourcesUsingLabels[PtrL interface {
+	*L
+	client.ObjectList
+}, PtrT interface {
+	*T
+	client.Object
+}, L any, T any](ctx context.Context, name string, cli client.Client, itemsFunc func(list PtrL) []T) ([]client.Object, error) {
+	listObj := PtrL(new(L))
+	err := cli.List(ctx, listObj,
+		client.MatchingLabels{
+			common.AppKubernetesNameLabel:      operandName,
+			common.AppKubernetesComponentLabel: operandComponent,
+			common.AppKubernetesManagedByLabel: common.AppKubernetesManagedByLabelValue,
+		},
+	)
+	if err != nil {
+		return nil, fmt.Errorf("error listing objects: %w", err)
+	}
+
+	// Filtering in a loop instead of using a FieldSelector in the List() call.
+	// It is only slightly inefficient, because all objects are already cached locally, so there is no API call.
```

Comment:
Again something for a follow up but doesn't this duplicate what the client would be doing when using FieldSelector anyway? I didn't think it would fire off additional API calls to filter by fields.

Reply:
Correct, but it was easier to implement here because of the internals of the client.

We use the cached client from the `controller-runtime` library. It watches and stores objects locally. 
If we want to use a field selector for a specific type, we first need to add an index for the field name into the cache. These indexes are separate for every type of object.

I think the code would look like this:
```go
const field = "metadata.name"
var indexerFunc client.IndexerFunc = func(object client.Object) []string {
	return []string{object.GetName()}
}

mgr.GetFieldIndexer().IndexField(ctx, &v1.ServiceAccount{}, field, indexerFunc)
mgr.GetFieldIndexer().IndexField(ctx, &v1.ConfigMap{}, field, indexerFunc)
mgr.GetFieldIndexer().IndexField(ctx, &v1.Service{}, field, indexerFunc)
mgr.GetFieldIndexer().IndexField(ctx, &appsv1.Deployment{}, field, indexerFunc)
mgr.GetFieldIndexer().IndexField(ctx, &routev1.Route{}, field, indexerFunc)
```
I haven't tested it, so maybe it's incorrect.
I didn't like that we need to register the index for each type separately.